### PR TITLE
Definition of the IO interface + HDF5/CGNS implementation

### PR DESCRIPTION
**Pull request by bmxam**

### CODE REVIEW

File: src/io/io_interface.jl [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+@show keys(result.data)
+```
+
+# Dev
+Possible names:
+* read_file
+* read_from_file
+* same with "load" instead of "read"
+
+Questions:
```
Comment (ghislainb) [inline src/io/io_interface.jl:39]: ?

**Review by ghislainb — APPROVED** [review]
nice work !

File: ext/HDF5Ext/common.jl [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    BCUBE_ENTITY_TO_CGNS_ENTITY[T]
+end
+
+cgns_entity_to_bcube_entity(code::Integer) = CGNS_ENTITY_TO_BCUBE_ENTITY[code]()
+
+function get_child(parent; name = "", type = "")
+    for child_name in keys(parent)
+        child = parent[child_name]
+        child_match(child, name, type) && (return child)
+    end
```
[the remark this answers is not in the document]
Comment (bmxam) [reply ext/HDF5Ext/common.jl:32]: > other suggestion (not tested):
> 
> ```julia
> get_child(parent; name = "", type = "") = findfirst(x->child_match(x, name, type), parent)
> ```

`findfirst` on a Dict returns the first "key" that fullfils the condition, but we want the value. So the function would look like
```julia
key = findfirst(x->child_match(x, name, type), parent)
return isnothing(key) ? nothing : parent[key]
```
do you prefer this?

File: ext/HDF5Ext/common.jl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+end
+
+function get_children(parent; name = "", type = "")
+    filtered =
+        filter(child_name -> child_match(parent[child_name], name, type), keys(parent))
+    map(child_name -> parent[child_name], filtered)
+end
```
[the remark this answers is not in the document]
Comment (bmxam) [reply ext/HDF5Ext/common.jl:39]: > suggestion (not tested):
> 
> ```julia
> get_children(parent; name = "", type = "") = findall(x->child_match(x, name, type), parent)
> ```

a solution would be:
```julia
keys = findall(x->child_match(x, name, type), parent)
return map(key -> parent[key], keys)
```
do you prefer this?

File: ext/HDF5Ext/common.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghislainb) [reply ext/HDF5Ext/common.jl:32]: yes !

**Pull request merged**